File: GraphBLAS/GraphBLAS/gb_isscalar.m
function s = gb_isscalar (G)
%GB_ISSCALAR determine if the matrix is a scalar.  Not user-callable.
% isscalar (G) is true for an m-by-n GraphBLAS matrix if m and n are 1.
% G is an opaque GraphBLAS struct or a built-in matrix.

% SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2026, All Rights Reserved.
% SPDX-License-Identifier: Apache-2.0

[m, n] = gb_size (G) ;
s = (m == 1) && (n == 1) ;

File: GraphBLAS/GraphBLAS/private/util/gb_is_scalar.c
//------------------------------------------------------------------------------
// gb_is_scalar: check if a GrB_Matrix is a scalar with a single entry
//------------------------------------------------------------------------------

// SuiteSparse:GraphBLAS, Timothy A. Davis, (c) 2017-2026, All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

//------------------------------------------------------------------------------

GrB_Info gb_is_scalar
(
    // output:
    bool *is_scalar,    // true if A is a 1-by-1 GrB_Matrix with 1 entry
    // input
    GrB_Matrix A,
    char err [ERRLEN]
)
{ 
    uint64_t nrows, ncols, nvals ;
    OK (GrB_Matrix_nrows (&nrows, A)) ;
    OK (GrB_Matrix_ncols (&ncols, A)) ;
    OK (GrB_Matrix_nvals (&nvals, A)) ;
    (*is_scalar) = (nrows == 1 && ncols == 1 && nvals == 1) ;
    return (GrB_SUCCESS) ;
}

